MinPlatformPkg/Test/Library/TestPointLib/MmTestPointCommunication.c

```diff
@@ -207,45 +207,42 @@ MmiHandlerTestPointCopyData (
 **/
 VOID
 MmTestPointMmiHandlerGetDataByOffset (
-  IN MMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET     *MmiHandlerTestPointParameterGetDataByOffset
+  IN VOID     *MmiHandlerTestPointParameterGetDataByOffset
   )
 {
-  MMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET    MmiHandlerTestPointGetDataByOffset;
+  MMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET    *MmiHandlerTestPointGetDataByOffset;
   VOID                                                   *Data;
   UINTN                                                  DataSize;
   EFI_STATUS                                             Status;
 
   Data = NULL;
 
-  CopyMem (
-    &MmiHandlerTestPointGetDataByOffset,
-    MmiHandlerTestPointParameterGetDataByOffset,
-    sizeof(MmiHandlerTestPointGetDataByOffset)
-    );
+  MmiHandlerTestPointGetDataByOffset = (MMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET *)MmiHandlerTestPointParameterGetDataByOffset;
 
   //
   // Sanity check
   //
-  if (!IsBufferOutsideMmValid((UINTN)MmiHandlerTestPointGetDataByOffset.DataBuffer, (UINTN)MmiHandlerTestPointGetDataByOffset.DataSize)) {
+  // MU_CHANGE - Removing Sanity check as we are now completely working within the comm buffer
+  /*if (!IsBufferOutsideMmValid((UINTN)&MmiHandlerTestPointParameterGetDataByOffset->Data[0], (UINTN)*Size)) {
     DEBUG((DEBUG_ERROR, "MmTestPointMmiHandlerGetDataByOffset: MmTestPoint get data in SMRAM or overflow!\n"));
-    MmiHandlerTestPointParameterGetDataByOffset->Header.ReturnStatus = (UINT64)(INT64)(INTN)EFI_ACCESS_DENIED;
+    MmiHandlerTestPointParameterGetDataByOffset->ReturnStatus = (UINT64)(INT64)(INTN)EFI_ACCESS_DENIED;
     goto Done;
-  }
+  }*/

```

Comment:
My understanding is that this code remains in the file commented to show the delta from upstream code with the `MU_CHANGE` tag. Is that correct? If so, I don't see `MU_CHANGE` tags around other changes.

Reply:
If you think it's unnecessary to leave in I'm okay with just deleting it here.  Alternatively, I can add `MU_CHANGE` tags to everything to everything if you think that's better for tracking the deltas.  I do think I should upstream this to EDK2 at some point.

Reply:
As long as it's consistent I don't mind. Technically, we should add `MU_CHANGE` tags per usual process, so I'd lean toward that.

Reply:
I added MU_CHANGE comments to the changes.